packages/internals/src/engine-commands/getDmmf.ts

```diff
@@ -82,395 +49,94 @@ ${detailsHeader} ${message}`
-      return { result, shouldRetry }
-    }),
-    TE.chainW(({ result, shouldRetry }) => {
-      if (shouldRetry) {
-        return TE.left({
-          type: 'retry' as const,
-          reason: 'Retrying after "Please wait until"',
-          timeout: 5000,
-        })
-      }
-
-      return TE.right({ result })
-    }),
-    TE.chainW(({ result }) => {
-      // necessary, as sometimes the query engine prints some other stuff
-      // TODO: identify when the binary does that.
-      const firstCurly = result.stdout.indexOf('{')
-      const stdout = result.stdout.slice(firstCurly)
-
-      return pipe(
-        E.tryCatch(
-          () => JSON.parse(stdout) as DMMF.Document,
-          (e) => ({
-            type: 'parse-json' as const,
-            reason: 'Unable to parse JSON',
-            error: e as Error,
-            result,
-          }),
-        ),
-        TE.fromEither,
-      )
-    }),
+      ),
+    ),
   )
 
-  const dmmfEither = await pipeline()
-
   if (E.isRight(dmmfEither)) {
-    debug('dmmf retrieved without errors in getDmmfBinary')
-
-    // Unlink only when no error occurs, as `sendPanic` might need the `tempDatamodelPath` later in case of errors.
-    await unlinkTempDatamodelPath(options, tempDatamodelPath)()
-
-    const { right: dmmf } = dmmfEither
-    return dmmf
+    debug('dmmf data retrieved without errors in getDmmf Wasm')
+    const { right: data } = dmmfEither
+    return Promise.resolve(data)
   }
 
   /**
-   * Check which error to throw (using Either.Right), or whether to retry the whole function (using Either.Left).
+   * Check which error to throw.
    */
-  const errorEither: E.Either<
-    {
-      type: 'retry'
-      reason: string
-      timeout: number
-    },
-    RustPanic | GetDmmfError
-  > = match(dmmfEither.left)
-    .with({ type: 'execa' }, (e) => {
+  const error = match(dmmfEither.left)
+    .with({ type: 'wasm-error' }, (e) => {
       debugErrorType(e)
 
-      // If the unlikely ETXTBSY event happens, try it at least once more
-      if (
-        e.error.message.includes('Command failed with exit code 26 (ETXTBSY)') &&
-        options.retry &&
-        options.retry > 0
-      ) {
-        return E.left({
-          type: 'retry' as const,
-          reason: 'Retrying because of error "ETXTBSY"',
-          timeout: 500,
-        })
-      }
-
       /**
-       * Capture and propagate possible Rust panics.
+       * Capture and propagate possible Wasm panics.
        */
-      if (isExecaErrorCausedByRustPanic(e.error)) {
+      if (isWasmPanic(e.error)) {
+        const wasmError = e.error
         const panic = new RustPanic(
-          /* message */ e.error.shortMessage,
-          /* rustStack */ e.error.stderr,
-          /* request */ 'query-engine get-dmmf', // TODO: understand which type it expects
-          ErrorArea.QUERY_ENGINE_BINARY_CLI,
-          /* schemaPath */ options.datamodelPath ?? tempDatamodelPath,
```

Comment:
`datamodelPath` seems to be gone but is still part of the options.
I'm unsure if we can remove it without breaking something, something to check maybe?

Reply:
We can check that in another PR

Reply:
we still need to understand if someone is using only datamodelPath , there it will break
quick search says it will break redwoodjs for example
https://github.com/search?q=repo%3Aredwoodjs%2Fredwood%20getDmmf%20datamodelPath&type=code

I don’t think we want to break redwoodjs (potentially others)
Something to consider, there are a few options but an easy one is to keep the API compatible?

Reply:
Fixed